lib/NavLink.js

```diff
@@ -45,7 +67,7 @@ NavLink = React.createClass({
             return;
         }
 
-        var href = this.props.href;
```

Comment:
This is the primary issue. `this.props.href` is undefined because we no longer add this prop for name based NavLinks.